homeassistant/components/zwave_js/discovery_data_template.py

```diff
@@ -187,6 +189,14 @@ def resolve_data(self, value: ZwaveValue) -> str | None:
 
         if value.command_class == CommandClass.METER:
             scale_type = get_meter_scale_type(value)
+            # We do this because even though these are energy scales, they don't meet
+            # the unit requirements for the energy device class.
+            if scale_type in (
+                ElectricScale.PULSE,
+                ElectricScale.KILOVOLT_AMPERE_HOUR,
```

Comment:
Is kVAh not sufficiently close to KWh for the energy device class?
Sorry if this has already been considered, I wasn't part of the discussion and don't have the history to check.
And if it's not, is `KILOVOLT_AMPERE_REACTIVE` (different to `KILOVOLT_AMPERE_REACTIVE_HOUR`) still OK for the power device class?

Reply:
Here are the available sensor device classes with required unit of measurement:
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
This dict tells you what units are acceptable for a given device class: https://github.com/home-assistant/core/blob/bb42eb11761176964ee1533256ac219896475f6d/homeassistant/components/sensor/recorder.py#L75-L108

So based on that, `KILOVOLT_AMPERE_REACTIVE` would not be ok for power

Reply:
Thanks. In that case `ElectricScale.KILOVOLT_AMPERE_REACTIVE` should also be in this list.

Reply:
As an aside, can the power factor be obtained from devices that report in kVAh, to convert to kWh?

Reply:
theoretically, but I don't know that the Z-Wave spec requires that a device reports both kVAh and power factor when one is provided. We'd have to add some special handling for this as well even if it was possible. Something we can look into if someone has a device that reports both and can provide us a state dump of the device so we can try to figure it out

Reply:
Also I would hope that if the device provides both kVAh and power factor, that it also provides kWh. In that case, the end result would be that you have three sensors, the kWh one with the energy device class, the kVAh one with no device class, and the power factor one with power factor device class

Reply:
We don't calculate state though. We just set what the device reports.